### PR TITLE
showCloseModal, and all other integrations

### PR DESCRIPTION
**Pull request by wimaan3**

### CODE REVIEW

File: apps/user-profile/src/components/common/Stepper/Stepper.js [hunk 1]
```diff
@@ -4,16 +4,23 @@ import {
...
+  StepperContainer,
+  CloseIcon
 } from './StyledStepper';
 import { Button, Typography } from '@devlaunchers/components/components/atoms';
 import chevronLeftImg from './../../../images/btn-chevron-left.svg';
 import chevronRightImg from './../../../images/btn-chevron-right.svg';
 import checkImg from './../../../images/Onboarding/stepper/check.svg';
 import RainbowBar from '../RainbowBar';
+import { useOnboardingDataContext } from '../../../context/OnboardingDataContext.js'
+// import onboardingActions from '../../../state/actions';
```
Comment (judeamos) [inline apps/user-profile/src/components/common/Stepper/Stepper.js:19]: Remove comments that are not needed

File: apps/user-profile/src/state/actions/onboardingActions.js [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+const HIDE_PLATFORM_ONBOARDING_MODAL = "HIDE_PLATFORM_ONBOARDING_MODAL";
+const SHOW_INTRODUCTION_MODAL = "SHOW_INTRODUCTION_MODAL";
+const HIDE_INTRODUCTION_MODAL = "HIDE_INTRODUCTION_MODAL";
+const HIDE_ALL_MODALS = "HIDE_ALL_MODALS";
+const SET_USERS_ROLE = "SET_USERS_ROLE";
+const SET_USERS_EXPERIENCE = "SET_USERS_EXPERIENCE";
+const SET_USERS_INTEREST = "SET_USERS_INTEREST";
+const REMOVE_USERS_INTEREST = "REMOVE_USERS_INTEREST";
+const HIDE_CLOSING_MODAL = "HIDE_CLOSING_MODAL";
+const SHOW_CLOSING_MODAL = "SHOW_CLOSING_MODAL";
```
Comment (judeamos) [inline apps/user-profile/src/state/actions/onboardingActions.js:13]: Any reason why you switch the values of each variable? e.g 
from:  `const SHOW_PLATFORM_ONBOARDING_MODAL = "Show platform onboarding modal.";` 
to: `const SHOW_PLATFORM_ONBOARDING_MODAL = "SHOW_PLATFORM_ONBOARDING_MODAL";`

File: apps/user-profile/src/state/reducers/onboardingReducer.js [hunk 3]
```diff
@@ -3,9 +3,11 @@ import { onboardingActions } from './../actions';
 import { featureFlags } from './../../utils/featureFlags';
 
 export const initialOnboardingState = {
-    showIntroductionModal: true,
-    showPlatformOnboardingModal: false,
-    user: {
+    
+    showIntroductionModal: false,
+    showPlatformOnboardingModal: true,
+    showCloseModal: false,
```
Comment (judeamos) [inline apps/user-profile/src/state/reducers/onboardingReducer.js:9]: Revert to the default setting when pushing to development branch.
`showIntroductionModal: true`
`showPlatformOnboardingModal: true,`

Comment (judeamos) [inline apps/user-profile/src/state/reducers/onboardingReducer.js:9]: Revert these to the default setting: 
`showIntroductionModal: true,`
`showPlatformOnboardingModal: false,`

File: apps/user-profile/src/state/reducers/onboardingReducer.js [hunk 4]
```diff
@@ -64,18 +70,22 @@ export const onboardingReducer = (state, action) => {
             }
         }
         case onboardingActions.SHOW_PLATFORM_ONBOARDING_MODAL: {
+            console.log("hello")
```
Comment (judeamos) [inline apps/user-profile/src/state/reducers/onboardingReducer.js:73]: remove console log

File: apps/user-profile/src/state/actions/onboardingActions.js (hunk 2 above)
Comment (judeamos) [inline apps/user-profile/src/state/actions/onboardingActions.js:13]: Any reason why you switch the values of each variable? e.g
from: `const SHOW_PLATFORM_ONBOARDING_MODAL = "Show platform onboarding modal.";`
to: `const SHOW_PLATFORM_ONBOARDING_MODAL = "SHOW_PLATFORM_ONBOARDING_MODAL";`

File: apps/user-profile/src/components/common/Stepper/Stepper.js [hunk 5]
```diff
@@ -106,6 +117,11 @@ export default function Stepper({ steps = stepsMockData, startIndex = 0 }) {
       stepsData[index].config.buttons.next.disabled : false);
   };
 
+  const showClosingModal = () => {
+    console.log("actions", onboardingActions);
```
Comment (judeamos) [inline apps/user-profile/src/components/common/Stepper/Stepper.js:121]: Remove console log

File: apps/user-profile/src/components/common/Stepper/Stepper.js [hunk 6]
```diff
@@ -208,9 +224,22 @@ export default function Stepper({ steps = stepsMockData, startIndex = 0 }) {
...
+                        <path id="Vector_2" d="M1.35541 5.95712e-05C1.52977 -0.000448724 1.70205 0.0325236 1.85958 0.0965524C2.01712 0.160581 2.15591 0.25404 2.26576 0.37006L7.90295 6.37006C8.07461 6.54899 8.16846 6.77344 8.16846 7.00506C8.16846 7.23669 8.07461 7.46113 7.90295 7.64006L2.06735 13.6401C1.86924 13.8443 1.58457 13.9727 1.27595 13.9971C0.967335 14.0215 0.660055 13.9398 0.421709 13.7701C0.183363 13.6003 0.0334778 13.3564 0.0050231 13.092C-0.0234307 12.8276 0.0718786 12.5643 0.269983 12.3601L5.48701 7.00006L0.445053 1.64006C0.302334 1.49327 0.211676 1.31453 0.183805 1.12499C0.155935 0.935437 0.192019 0.743012 0.287788 0.570485C0.383557 0.397957 0.535003 0.252546 0.724205 0.151455C0.913407 0.0503649 1.13245 -0.00217204 1.35541 5.95712e-05Z" fill="#FFF"/>
```
Comment (judeamos) [inline apps/user-profile/src/components/common/Stepper/Stepper.js:238]: Can we move all of this into the image folder, and then import it into here. You can use the btn-chevron-left.svg as an example: 
image

File: apps/user-profile/src/components/common/Stepper/StyledStepper.js [hunk 7]
```diff
@@ -14,15 +18,24 @@ export const StepperHeader = styled.div`
     background: black;
     max-height: 120px;
     justify-content: center;
+    border: 10px red;
```
Comment (judeamos) [inline apps/user-profile/src/components/common/Stepper/StyledStepper.js:21]: is this needed?

File: apps/user-profile/src/components/modules/UserOnboardingModal/CloseModal/CloseModal.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+                        <path id="Vector_2" d="M1.35541 5.95712e-05C1.52977 -0.000448724 1.70205 0.0325236 1.85958 0.0965524C2.01712 0.160581 2.15591 0.25404 2.26576 0.37006L7.90295 6.37006C8.07461 6.54899 8.16846 6.77344 8.16846 7.00506C8.16846 7.23669 8.07461 7.46113 7.90295 7.64006L2.06735 13.6401C1.86924 13.8443 1.58457 13.9727 1.27595 13.9971C0.967335 14.0215 0.660055 13.9398 0.421709 13.7701C0.183363 13.6003 0.0334778 13.3564 0.0050231 13.092C-0.0234307 12.8276 0.0718786 12.5643 0.269983 12.3601L5.48701 7.00006L0.445053 1.64006C0.302334 1.49327 0.211676 1.31453 0.183805 1.12499C0.155935 0.935437 0.192019 0.743012 0.287788 0.570485C0.383557 0.397957 0.535003 0.252546 0.724205 0.151455C0.913407 0.0503649 1.13245 -0.00217204 1.35541 5.95712e-05Z" fill="#1C1C1C"/>
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/UserOnboardingModal/CloseModal/CloseModal.js:35]: Same again, move this to the image folder. If this is the same as the other svg, then you can reuse it.

File: apps/user-profile/src/components/modules/UserOnboardingModal/CloseModal/CloseModal.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
+import { IntroductionContainer, Header, Body, Footer } from "./StyledCloseModal";
+import { Typography } from "@devlaunchers/components/components/atoms";
+import Button from '@devlaunchers/components/components/atoms/Button';
+import RainbowBar from './../../../common/RainbowBar/RainbowBar';
+
+import { useOnboardingDataContext } from './../../../../context/OnboardingDataContext';
+import { onboardingActions } from './../../../../state/actions';
+import { LineIcon } from "react-share";
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/UserOnboardingModal/CloseModal/CloseModal.js:8]: Whats this used for?

File: apps/user-profile/src/components/modules/UserOnboardingModal/CloseModal/CloseModal.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    const onSetUpLater = () => {
+        dispatch({ type: onboardingActions.HIDE_ALL_MODALS });
+    }
+    const quitCloseModal = () => {
+        dispatch({ type: onboardingActions.HIDE_CLOSING_MODAL });
+    }
+
+    return (
+        <>
+            <IntroductionContainer>
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/UserOnboardingModal/CloseModal/CloseModal.js:30]: Should'nt this styled component change to fit the name of the closing modal?

File: apps/user-profile/src/components/modules/UserOnboardingModal/CloseModal/CloseModal.js [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * @description This is custom modal for the user onboarding. 
+ */
+export default function CloseModal() {
+    const { onboardingData, dispatch } = useOnboardingDataContext();
+
+    const onGetStarted = () => {
+        dispatch({ type: onboardingActions.SHOW_PLATFORM_ONBOARDING_MODAL });
+    }
+    const onSetUpLater = () => {
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/UserOnboardingModal/CloseModal/CloseModal.js:21]: can we use a better name to describe this action.

File: apps/user-profile/src/components/modules/UserOnboardingModal/CloseModal/CloseModal.js [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
... 25.0001V15.1301H58.1871L64.1371 23.0681V15.1301H65.2011V25.0001H64.3331L58.3691 17.0341V25.0001H57.2911ZM73.6044 25.1261C72.6337 25.1261 71.803 24.9207 71.1124 24.5101C70.4217 24.0994 69.8897 23.5161 69.5164 22.7601C69.1524 22.0041 68.9704 21.1034 68.9704 20.0581C68.9704 19.0221 69.1524 18.1261 69.5164 17.3701C69.8897 16.6141 70.4217 16.0307 71.1124 15.6201C71.803 15.2094 72.6337 15.0041 73.6044 15.0041C74.2857 15.0041 74.911 15.1114 75.4804 15.3261C76.059 15.5407 76.549 15.8534 76.9504 16.2641L76.5024 17.2021C76.073 16.8007 75.625 16.5114 75.1584 16.3341C74.701 16.1474 74.1924 16.0541 73.6324 16.0541C72.5217 16.0541 71.6724 16.4041 71.0844 17.1041C70.4964 17.8041 70.2024 18.7887 70.2024 20.0581C70.2024 21.3367 70.4964 22.3261 71.0844 23.0261C71.6724 23.7261 72.5217 24.0761 73.6324 24.0761C74.1924 24.0761 74.701 23.9874 75.1584 23.8101C75.625 23.6234 76.073 23.3294 76.5024 22.9281L76.9504 23.8661C76.549 24.2674 76.059 24.5801 75.4804 24.8041C74.911 25.0187 74.2857 25.1261 73.6044 25.1261ZM80.4343 25.0001V15.1301H86.6643V16.0961H81.5543V19.5121H86.3703V20.4781H81.5543V24.0341H86.6643V25.0001H80.4343ZM90.2764 25.0001V15.1301H91.4244V23.9921H96.4504V25.0001H90.2764Z" fill="#295774"/>
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/UserOnboardingModal/CloseModal/CloseModal.js:69]: You are using SVG images in the button, are these meant to be text?

File: apps/user-profile/src/components/modules/UserOnboardingModal/IntroductionModal/IntroductionModal.js [hunk 13]
```diff
@@ -13,11 +13,18 @@ import { onboardingActions } from './../../../../state/actions';
  */
 export default function IntroductionModal() {
     const { onboardingData, dispatch } = useOnboardingDataContext();
-
+    if (!onboardingData || onboardingData.showIntroductionModal === false) {
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/UserOnboardingModal/IntroductionModal/IntroductionModal.js:16]: Can you move this logic to the `UserOnboardingModal.js` file, its a better place to have this implementation.

File: apps/user-profile/src/components/modules/UserOnboardingModal/IntroductionModal/IntroductionModal.js [hunk 14]
```diff
@@ -13,11 +13,18 @@ import { onboardingActions } from './../../../../state/actions';
...
+        console.log("hello")
+        return null; //dont render the modal if showIntroductionModal is false
+    }
     const onGetStarted = () => {
+        console.log("hello")
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/UserOnboardingModal/IntroductionModal/IntroductionModal.js:21]: remove console.logs

File: apps/user-profile/src/components/modules/UserOnboardingModal/UserOnboardingModal.js [hunk 15]
```diff
@@ -2,16 +2,79 @@ import { useState, useEffect } from "react";
...
+//                 style={userUnboardingModalStyle}
+//                 contentLabel="User Onboarding"
+//             >
+//                 <ModalContainer>
+//                     {showModals()}
+//                 </ModalContainer>
+//             </Modal>
+//         </>
+//     );
+// }
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/UserOnboardingModal/UserOnboardingModal.js:61]: remove comments that are not needed

File: apps/user-profile/src/components/modules/UserOnboardingModal/UserOnboardingModal.js [hunk 16]
```diff
@@ -2,16 +2,79 @@ import { useState, useEffect } from "react";
...
+export default function UserOnboardingModal({modalsToShow}) {
+    const { onboardingData, dispatch } = useOnboardingDataContext();
+    const tempData = useOnboardingDataContext();
+    const [modalIsOpen, setModalIsOpen] = useState(false);
+
+    useEffect(() => {
+        console.log('data', tempData)
+        console.log('Modal conditions:', onboardingData?.showIntroductionModal, onboardingData?.showPlatformOnboardingModal);
+        console.log('HELLO', onboardingData)
+        console.log('MODALSTOSHOW', modalsToShow)
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/UserOnboardingModal/UserOnboardingModal.js:71]: remove console.logs that are not needed

File: apps/user-profile/src/components/modules/UserOnboardingModal/UserOnboardingModal.js [hunk 17]
```diff
@@ -2,16 +2,79 @@ import { useState, useEffect } from "react";
...
+//                 <ModalContainer>
+//                     {showModals()}
+//                 </ModalContainer>
+//             </Modal>
+//         </>
+//     );
+// }
+export default function UserOnboardingModal({modalsToShow}) {
+    const { onboardingData, dispatch } = useOnboardingDataContext();
+    const tempData = useOnboardingDataContext();
```
Comment (judeamos) [inline apps/user-profile/src/components/modules/UserOnboardingModal/UserOnboardingModal.js:64]: is this needed?

**Review by judeamos — CHANGES_REQUESTED** [review]
I've left comments and feedback, let me know if you need any help.

File: apps/user-profile/src/state/actions/onboardingActions.js (hunk 2 above)
Comment (wimaan3) [reply apps/user-profile/src/state/actions/onboardingActions.js:13]: I was just testing to see if all capital letters would change anything for the old errors we were having.

File: apps/user-profile/src/components/common/Stepper/StyledStepper.js (hunk 7 above)
Comment (wimaan3) [reply apps/user-profile/src/components/common/Stepper/StyledStepper.js:21]: I think so?

**Comment by wimaan3** [comment]
wrong PR

**Pull request closed without merging**